packages/blocks/src/api/parser/index.js

```diff
@@ -21,49 +21,19 @@ import { getBlockAttributes } from './get-block-attributes';
 import { applyBlockDeprecatedVersions } from './apply-block-deprecated-versions';
 import { applyBuiltInValidationFixes } from './apply-built-in-validation-fixes';
 
-/**
- * The raw structure of a block includes its attributes, inner
- * blocks, and inner HTML. It is important to distinguish inner blocks from
- * the HTML content of the block as only the latter is relevant for block
- * validation and edit operations.
- *
- * @typedef WPRawBlock
- *
- * @property {string=}         blockName    Block name
```

Comment:
Handle instances where a raw block doesn't have a `blockName` in those special cases, rather than have it as optional. Currently there are lots of points `blockName` is accessed without checking for `undefined` so I assume in most cases it is defined. Opinions or advice?

Reply:
this occurs when there's freeform HTML within a Gutenberg serialized post. those blocks retain their identity through the editor cycle, so even when re-serializing a post on save they will have no `blockName` (because they aren't inside a block)

if we do this we'll need to discuss making changing and breaking changes to the "non-block content" semanatics, which would have to happen first.

now, with TypeScript we _can_ apply some type magic to remove the optionality of that field, but only in places where we've already confirmed that it's there, such as inside of `block-list/block.js`

Reply:
@dmsnell I am working hard to only type the existing code during this initial step, not modify how things work. I agree, leaning on TypeScript to remove or add optionality of properties where needed makes the most sense.

Reply:
I'll start by doing it on a case by case basis and we'll see if any patterns emerge which can be extracted

Reply:
> not modify how things work

Thank you. I only meant to imply that doing what you were asking would require changing the project, something that's out of scope for a typing PR.

There is opportunity in `block-list/block` to do what you are wanting purely through types, but I don't think it'll be fruitful here for now.

Reply:
@dmsnell can you add link to your reference of `block-list/block`?

Reply:
@johnhooks it's the `packages/block-editor/src/components/block-list/block.js` file